search-commons/src/main/java/no/unit/nva/search/CognitoAuthenticator.java

```diff
@@ -0,0 +1,109 @@
+package no.unit.nva.search;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.jr.ob.JSON;
+import no.unit.nva.auth.CognitoCredentials;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.paths.UriWrapper;
+
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Objects;
+
+import static com.amazonaws.auth.internal.SignerConstants.AUTHORIZATION;
+import static no.unit.nva.auth.AuthorizedBackendClient.APPLICATION_X_WWW_FORM_URLENCODED;
+import static nva.commons.core.attempt.Try.attempt;
+import static org.apache.http.protocol.HTTP.CONTENT_TYPE;
+
+public class CognitoAuthenticator {
+    private final CognitoCredentials credentials;
+    public static final String AUTHORIZATION_ERROR_MESSAGE = "Could not authorizer client";
+    public static final String GRANT_TYPE_CLIENT_CREDENTIALS = "grant_type=client_credentials";
+    public static final String JWT_TOKEN_FIELD = "access_token";
+
+    private final HttpClient httpClient;
+
+    public CognitoAuthenticator(HttpClient httpClient, CognitoCredentials credentials) {
+        this.httpClient = httpClient;
+        this.credentials = credentials;
+    }
+
+    @JacocoGenerated
+    public static CognitoAuthenticator prepareWithCognitoCredentials(CognitoCredentials cognitoCredentials) {
+        return prepareWithCognitoCredentials(HttpClient.newHttpClient(), cognitoCredentials);
+    }
+
+    public static CognitoAuthenticator prepareWithCognitoCredentials(HttpClient httpClient,
+                                                                     CognitoCredentials cognitoApiClientCredentials) {
+        return new CognitoAuthenticator(httpClient, cognitoApiClientCredentials);
+    }
+
+    private String getBearerTokenString() {
```

Comment:
Probably 

```
    private HttpRequest createTokenRequest() {
        var tokenUri = standardOauth2TokenEndpoint(credentials.getCognitoOAuthServerUri());
        return formatRequestForJwtToken(tokenUri);
    }
```
and then

```
    private HttpResponse<String> fetchToken() {
        return attempt(
                () -> this.httpClient.send(createTokenRequest(), HttpResponse.BodyHandlers.ofString(StandardCharsets.UTF_8))
        )
                .map(this::responseIsSuccessful)
                .orElseThrow();
    }
```

Which leaves this method as:

```
        return attempt(() -> fetchToken())
                .map(HttpResponse::body)
                .map(JSON.std::mapFrom)
                .map(json -> json.get(JWT_TOKEN_FIELD))
                .toOptional()
                .map(Objects::toString)
                .orElseThrow();
```

that is to say: inline `sendRequestAndExtractToken()`